bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FtpFolderWatcherHandler.java

```diff
@@ -0,0 +1,271 @@
+                }
+            }
+        }
+        return dirList;
+    }
+
+    private void connectionKeepAlive() {
+        if (ftp.isConnected() == false) {
+            if (config.secureMode.equals("NONE")) {
+                ftp = new FTPClient();
+            } else {
+                switch (config.secureMode) {
+                    case "NONE":
+                        ftp = new FTPClient();
+                        break;
+                    case "IMPLICIT":
+                        ftp = new FTPSClient(true);
+                        break;
+                    case "EXPLICIT":
+                        ftp = new FTPSClient(false);
+                        break;
+                }
+            }
+
+            int reply = 0;
+            ftp.setListHiddenFiles(config.listHidden);
+            ftp.setConnectTimeout(config.connectionTimeout * 1000);
+
+            try {
+                if (config.ftpPort > 0) {
+                    ftp.connect(config.ftpAddress, config.ftpPort);
+                } else {
+                    ftp.connect(config.ftpAddress);
+                }
+                reply = ftp.getReplyCode();
+
+                if (!FTPReply.isPositiveCompletion(reply)) {
+                    ftp.disconnect();
+                    logger.debug("FTP server refused connection.");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "FTP server refused connection.");
+                    return;
+                }
+            } catch (IOException e) {
+                if (ftp.isConnected()) {
+                    try {
+                        ftp.disconnect();
+                    } catch (IOException e2) {
+                        logger.debug("Error disconneting, lost connection? : ", e2);
+                    }
+                }
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                return;
+            }
+            try {
+                if (!ftp.login(config.ftpUsername, config.ftpPassword)) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ftp.getReplyString());
+                    ftp.logout();
+                    return;
+                }
+                updateStatus(ThingStatus.ONLINE);
+                ScheduledFuture<?> executionJob = this.executionJob;
+                if (executionJob != null) {
+                    executionJob.cancel(true);
+                }
+                this.executionJob = scheduler.scheduleWithFixedDelay(this::refreshFTPFolderInformation, 0,
+                        config.pollInterval, TimeUnit.SECONDS);
+                return;
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+            return;
```

Comment:
These return statements aren't needed since you are already at the end of the method.
```suggestion
            } catch (IOException e) {
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
            }
```

Reply:
Fixed